### PR TITLE
[WIP] Port FFmpeg/libav* to Serenity!

### PR DESCRIPTION
**Pull request by 06needhamt**
This PR contains my current efforts to port FFmpeg/libav* to SerenityOS. 

This is the first step to me achieving my dream of implementing a simple audio/video player for Serenity. In the end, I would like to have all the video decoding done natively without any external dependencies.

The Port is not currently functional due to some issues with format strings which I hope to resolve soon.

Any comments on my work so far would be greatly appreciated

### CODE REVIEW

File: Ports/FFmpeg/package.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+makeopts="HOSTCC=${CC} HOSTLD=${CC}"
+depends="SDL2"
+
+configure() {
+	run ./configure --enable-cross-compile --target-os=serenity --arch=i686 \
+	    --enable-ffmpeg --enable-ffprobe --enable-ffplay --cross-prefix=${CC%gcc} \
+	    --disable-txtpages --disable-manpages --disable-podpages \
+	    --disable-static --enable-shared \
+	    --enable-debug --disable-stripping # Remove Before PR!
+}
```
Comment (bcoles) [inline Ports/FFmpeg/package.sh:16]: You should be able to move all of these into a `configopts` parameter.

```suggestion
configopts="--enable-cross-compile --enable-ffmpeg --enable-ffprobe --enable-ffplay --cross-prefix=${CC%gcc} --disable-txtpages --disable-manpages --disable-podpages --disable-static --enable-shared --enable-debug --disable-stripping"
```

`configopts` is already defined in `.port_include.sh` and sets the `host` flag automatically:

```
--host="${SERENITY_ARCH}-pc-serenity"
```

I think you can also drop the ` --target-os=serenity --arch=i686` flags. This should be picked up by the configure script automatically (maybe).

Also, unless generating the man pages is causing issues, you can remove `--disable-manpages`. It doesn't hurt to have the man pages installed. Many other ports do this, and maybe we'll one day have a port of `man` to read man pages in groff format.

File: Userland/Libraries/LibDebug/Dwarf/DIE.cpp [hunk 2]
```diff
@@ -139,6 +139,22 @@ DIE::AttributeValue DIE::get_attribute_value(AttributeDataForm form,
...
+        break;
+    }
+    case AttributeDataForm::Ref8: {
+        u64 data;
+        debug_info_stream >> data;
+            VERIFY(!debug_info_stream.has_any_error());
+        value.type = AttributeValue::Type::DieReference;
+        value.data.as_u32 = data + m_compilation_unit.offset(); // FIXME: Do we need to properly handle 64-Bit values?
+        break;
+    }
```
Comment (bcoles) [inline Userland/Libraries/LibDebug/Dwarf/DIE.cpp:157]: > FIXME: Do we need to properly handle 64-Bit values?

No, we don't. #6036

If you're interested in implementing this, there's a patch here which adds support for 64 bit values to LibDebug.

https://github.com/SerenityOS/serenity/pull/6287#issuecomment-818505818

However, I found that this generated corrupted ELF crash dump files, so I never merged it investigated further.

File: Userland/Libraries/LibC/sys/socket.cpp [hunk 3]
```diff
@@ -112,6 +112,12 @@ ssize_t recvfrom(int sockfd, void* buffer, size_t buffer_length, int flags, stru
     return rc;
 }
 
+int getnameinfo(const struct sockaddr*, socklen_t, char*, socklen_t, char*, socklen_t, int)
+{
+    // FIXME: Actually implement this
```
Comment (Nicholas-Baron) [inline Userland/Libraries/LibC/sys/socket.cpp:117]:
```suggestion
    VERIFY_NOT_REACHED();
```
More like this?

Comment (linusg) [reply Userland/Libraries/LibC/sys/socket.cpp:117]: if anything, more like this:

```suggestion
    TODO();
```

**Pull request closed without merging**